src/ui.jsx

```diff
@@ -0,0 +1,392 @@
+/**
+ * ui.jsx
+ *
+ * A file describing the UI of the application using React. A lot of the state accesses occur on the state property
+ * which apparently isn't the best practice but it gets rid of lots of unnecessary re-renders.
+ *
+ */
+"use strict";
+/*eslint-disable no-console, no-underscore-dangle*/
```

Comment:
Throw directives like this in an `.eslintrc` file so you don't have to repeat them throughout. You can check this into source control so everyone on the project would stick to a common style. ESLint will load this file automatically.